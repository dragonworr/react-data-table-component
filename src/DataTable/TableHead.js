import styled, { withTheme } from 'styled-components';

const TableHead = styled.thead`
  text-align: left;
  background-color: ${props => props.theme.header.backgroundColor};
`;

export default withTheme(TableHead);
