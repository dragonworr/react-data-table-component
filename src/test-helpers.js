import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import theme from './themes/default';

export const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = tree => {
  const context = mount(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return mount(tree, { context });
};
