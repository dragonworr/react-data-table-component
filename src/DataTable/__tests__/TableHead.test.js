import 'jest-styled-components';

import React from 'react';
import TableHead from '../TableHead';
import { shallowWithTheme } from '../../test-helpers';

test('component <TableHead /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableHead />);
  expect(wrapper.dive()).toMatchSnapshot();
});
