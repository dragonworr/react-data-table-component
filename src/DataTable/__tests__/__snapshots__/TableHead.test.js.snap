// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <TableHead /> should render correctly 1`] = `
.c0 {
  text-align: left;
  background-color: transparent;
}

<thead
  className="c0"
/>
`;
